lib/jwt/claims_validator.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require_relative './error'
+
+module JWT
+  class ClaimsValidator
+    def initialize(payload)
+      @payload = payload.each_with_object({}) { |(k, v), h| h[k.to_sym] = v }
+    end
+
+    def validate
+      validate_exp if @payload[:exp]
```

Comment:
You have to check that the key actually is present in the payload, this row is now checking if the value returned from the hash is thruthy. 

I think this will not validate cases where exp is false or nil eg:
```
{ exp: false }
```